src/figuresClasses.ts

```diff
@@ -1,19 +1,116 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Color {
+  Red = 'triangle',
+  Green = 'circle',
+  Blue = 'rectangle',
+}
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+abstract class BaseForFigure implements Figure {
+  abstract readonly shape: Shape;
+
+  abstract color: Color;
 
+  abstract getArea(): number;
+
+  protected checkAllSides(sides: number[]): void {
+    if (sides.some((side) => side <= 0)) {
+      throw new Error(`${this.shape} should have only positive params`);
+    }
+  }
+
+  static roundNumber(value: number): number {
+    return Math.floor(value * 100) / 100;
+  }
 }
 
-export class Circle {
+export class Triangle extends BaseForFigure {
+  readonly shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    super();
+
+    this.checkAllSides([a, b, c]);
+
+    const [
+      sideA,
+      sideB,
+      sideC,
+    ]: number[] = [a, b, c].sort((prev, next) => prev - next);
 
+    if (sideC >= sideA + sideB) {
+      throw new Error('Impossible to create a triangle with these sides');
+    }
+  }
+
+  getArea(): number {
+    const semiP = (this.a + this.b + this.c) / 2;
+
+    const area = Math.sqrt(
+      semiP
+      * (semiP - this.a)
+      * (semiP - this.b)
+      * (semiP - this.c),
+    );
```

Comment:
This is not a mistake, just a suggestion to use destructurization
```suggestion
    const { a, b, c } = this;
    const semiP = (a + b + c) / 2;

    const area = Math.sqrt(
      semiP
      * (semiP - a)
      * (semiP - b)
      * (semiP - c),
    );
```